### PR TITLE
 RavenDB-11139: Fix schema upgrade.

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

**Review by grisha-kotler — APPROVED** [review]

File: src/Raven.Server/Storage/Schema/Updates/Server/From14.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+namespace Raven.Server.Storage.Schema.Updates.Server
+{
+    public unsafe class From14 : ISchemaUpdate
+    {
+        public bool Update(UpdateStep step)
+        {
+            var dbs = new List<string>();
+            const string dbKey = "db/";
+
+            var oldIdentitiesSchema = new TableSchema().
```
Comment (ppekrol) [inline src/Raven.Server/Storage/Schema/Updates/Server/From14.cs]: At this point we shouldn't have the 'old schema' ?

File: src/Voron/Data/Tables/TableSchema.cs [hunk 2]
```diff
@@ -89,7 +89,7 @@ public class SchemaIndexDef
             /// This is relevant only for migration purposes and should never be set for non migration code.
             /// It is meant to allow deleting an item when we create a new index that isn't in the storage
             /// </summary>
-            public bool Dangerous_IgnoreForDeletes;
+            public bool Dangerous_IgnoreForDeletesAndMissingValues;
```
Comment (arekpalinski) [inline src/Voron/Data/Tables/TableSchema.cs:92]: News to update comment to explain what 'AndMissingValues' means

File: src/Raven.Server/Storage/Schema/Updates/Server/From14.cs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+        private (BlittableJsonReaderObject, long) GetBjroAndIndex(JsonOperationContext context, Table items, Slice lowerKeySlice)
+        {
+            if (items.ReadByKey(lowerKeySlice, out TableValueReader reader) == false)
+                return (null, 0L);
+
+            var ptr = reader.Read(2, out int size);
+            var bjro = new BlittableJsonReaderObject(ptr, size, context);
+
+            var index = *(long*)reader.Read(3, out var sizeOfLong);
```
Comment (arekpalinski) [inline src/Raven.Server/Storage/Schema/Updates/Server/From14.cs]: Can we replace 2 and 3 by constants from the relevant schema?

File: src/Raven.Server/Storage/Schema/Updates/Server/From13.cs [hunk 4]
```diff
@@ -112,19 +113,26 @@ public bool Update(UpdateStep step)
...
                         foreach (var item in readIdentitiesTable.SeekByPrimaryKeyPrefix(keyPrefix, Slices.Empty, 0))
                         {
-                            var index = TableValueToLong((int)ClusterStateMachine.IdentitiesTable.Index, ref item.Value.Reader);
-                            var value = TableValueToLong((int)ClusterStateMachine.IdentitiesTable.Value, ref item.Value.Reader);
+                            var indexPtr = item.Value.Reader.Read((int)ClusterStateMachine.IdentitiesTable.Index, out var size);
 
-                            using (GetPrefixIndexSlices(step.ReadTx.Allocator, db, index, out var buffer))
-                            using (Slice.External(step.WriteTx.Allocator, buffer.Ptr, buffer.Length, out var prefixIndexSlice))
-                            using (writeIdentitiesTable.Allocate(out TableValueBuilder write))
+                            // we come from RC1
```
Comment (arekpalinski) [inline src/Raven.Server/Storage/Schema/Updates/Server/From13.cs]: 4.2 RC1

File: src/Voron/Data/Tables/TableSchema.cs (hunk 2 above)
Comment (garayx) [reply src/Voron/Data/Tables/TableSchema.cs:92]: done

File: src/Raven.Server/Storage/Schema/Updates/Server/From14.cs (hunk 3 above)
Comment (garayx) [reply src/Raven.Server/Storage/Schema/Updates/Server/From14.cs]: we dont have an appropriate enum for items schema, so I just put a comment

File: src/Raven.Server/Storage/Schema/Updates/Server/From13.cs (hunk 4 above)
Comment (garayx) [reply src/Raven.Server/Storage/Schema/Updates/Server/From13.cs]: done

File: src/Raven.Server/Storage/Schema/Updates/Server/From14.cs (hunk 3 above)
Comment (arekpalinski) [reply src/Raven.Server/Storage/Schema/Updates/Server/From14.cs]: okay, anything more meaningful will work

Comment (ayende) [reply src/Raven.Server/Storage/Schema/Updates/Server/From14.cs]: Constants with a comment would be best, yes.

**Review by ayende — APPROVED** [review]

**Pull request merged**